src/main/kotlin/app/revanced/patches/shared/mapping/misc/patch/ResourceMappingPatch.kt

```diff
@@ -71,3 +74,14 @@ class ResourceMappingPatch : ResourcePatch {
 }
 
 data class ResourceElement(val type: String, val name: String, val id: Long)
+
+/**
+ * @return the first constant instruction with the resource id, or -1 if not found.
+ */
+fun Method.indexOfFirstConstantInstruction(constantValue: Long): Int {
+    return implementation?.let {
+        it.instructions.indexOfFirst { instruction ->
+            instruction.opcode == Opcode.CONST && (instruction as WideLiteralInstruction).wideLiteral == constantValue
+        }
+    } ?: -1;
+}
```

Comment:
I don't think placing methods mainly interacting with bytecode in a resource patches file is a good idea. Instead, create a utils class for it if it does not exist, either in the patch package or for all patches, or add the function to a companion object of the patch where you use this function.